1_fetch/src/get_sb_data.R

```diff
@@ -217,4 +217,45 @@ download_children <-function(sites, sb_table_reduced, dldir, workdir, outdir, ou
   write_csv(itemfails, filepath2)
   return_df = c(filepath1, filepath2)
   return(return_df)
+}
+
+calc_avg_monthly_weather <- function(sb_data) {
+  sb_data <- read_csv(sb_data, show_col_types = FALSE) %>%
+    suppressMessages()
+  precip <- sb_data %>%
+    select(COMID, contains("_PPT_")) %>%
+    pivot_longer(!COMID, names_to = "unit_month_year", values_to = "precip") %>%
+    mutate(unit = str_sub(unit_month_year, 1, 3), 
+           month_name = str_sub(unit_month_year, 9, 11)) %>%
+    group_by(COMID, unit, month_name) %>%
+    summarise(avg_monthly_ppt = mean(precip, na.rm = TRUE), .groups = "drop") %>%
+    mutate(month_num = case_when(month_name == "JAN" ~ 1, month_name == "FEB" ~ 2, 
+                                 month_name == "MAR" ~ 3, month_name == "APR" ~ 4, 
+                                 month_name == "MAY" ~ 5, month_name == "JUN" ~ 6, 
+                                 month_name == "JUL" ~ 7, month_name == "AUG" ~ 8, 
+                                 month_name == "SEP" ~ 9, month_name == "OCT" ~ 10, 
+                                 month_name == "NOV" ~ 11, month_name == "DEC" ~ 12),
+           name = paste0(unit, "_PPT_", month_name)) %>%
+    arrange(COMID, unit, month_num) %>%
+    select(COMID, name, avg_monthly_ppt) %>%
+    pivot_wider(names_from = "name", values_from = "avg_monthly_ppt")
+  temperature <- sb_data %>%
```

Comment:
It looks like a lot is repeated for temperature. Could make a function with arguments for name ("_TAV_"), values_to = "temp", column to use for mean, and the new column name (avg_monthly_temp)